clio-server/src/test/scala/org/broadinstitute/clio/server/webservice/WgsCramWebServiceSpec.scala

```diff
@@ -0,0 +1,200 @@
+package org.broadinstitute.clio.server.webservice
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Route
+import com.sksamuel.elastic4s.searches.queries.BoolQueryDefinition
+import io.circe.Json
+import org.broadinstitute.clio.server.MockClioApp
+import org.broadinstitute.clio.server.dataaccess.MemorySearchDAO
+import org.broadinstitute.clio.server.dataaccess.elasticsearch.DocumentWgsCram
+import org.broadinstitute.clio.server.service.WgsCramService
+import org.broadinstitute.clio.transfer.model.WgsCramIndex
+import org.broadinstitute.clio.transfer.model.wgscram.TransferWgsCramV1QueryInput
+import org.broadinstitute.clio.util.model.{DocumentStatus, Location, UpsertId}
+
+class WgsCramWebServiceSpec extends BaseWebserviceSpec {
+  behavior of "WgsCramWebService"
+
+  it should "postMetadata with OnPrem location" in {
+    val webService = new MockWgsCramWebService()
+    Post("/metadata/OnPrem/proj0/sample_alias0/1", Map("notes" -> "some note")) ~> webService.cramPostMetadata ~> check {
```

Comment:
If we can refactor our endpoint code, so that the only difference between different index endpoints is the key path formation and the service API to use, we can get rid of this duplicated test code.